stubs/tensorflow/tensorflow/keras/metrics.pyi

```diff
@@ -1,9 +1,136 @@
-from tensorflow import Tensor
-from tensorflow._aliases import TensorCompatible
+from _typeshed import Incomplete
+from abc import ABCMeta, abstractmethod
+from collections.abc import Callable, Iterable, Sequence
+from typing import Any, Literal
+from typing_extensions import Self, TypeAlias, override
 
+import tensorflow as tf
+from tensorflow import Operation, Tensor
+from tensorflow._aliases import DTypeLike, KerasSerializable, TensorCompatible
+from tensorflow.keras.initializers import _Initializer
+from tensorflow.keras.layers import _Constraint
+from tensorflow.keras.regularizers import _Regularizer
+
+_Output: TypeAlias = Tensor | dict[str, Tensor]
+
+class Metric(tf.keras.layers.Layer[tf.Tensor, tf.Tensor], metaclass=ABCMeta):
+    def __init__(self, name: str | None = None, dtype: DTypeLike | None = None) -> None: ...
+    def __new__(cls, *args: Any, **kwargs: Any) -> Self: ...
+    def __call__(self, *args: Incomplete, **kwargs: Incomplete) -> Incomplete: ...
+    def __deepcopy__(self, memo: Incomplete | None = None) -> Incomplete: ...
```

Comment:
Why `__deepcopy__`, `__new__`, and `__call__`? call should be inherited from layer class. deepcopy seems weird to define here. Unsure what special thing new is for either.

Reply:
Removed in d45a599e9